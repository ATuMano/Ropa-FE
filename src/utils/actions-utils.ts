export const createActionTypeRunner = (actionType: string) => ({
  INIT: `${actionType}_INIT`,
  SUCCESS: `${actionType}_SUCCESS`,
  ERROR: `${actionType}_ERROR`
});
