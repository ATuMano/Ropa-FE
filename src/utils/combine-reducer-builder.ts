import { State } from "app/store/types";
export type ReducerNamed = {
  reducer: any;
  name: (obj: State) => any;
};
