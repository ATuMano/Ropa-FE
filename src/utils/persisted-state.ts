import { State } from "app/store/types";

export const PERSISTED_STATE = "state";

type persistedState = {
  [key: string]: any;
};

const whiteList: string[] = [];

export const getPersistedState: () => any = () => {
  try {
    const serializedState = window.localStorage.getItem(PERSISTED_STATE);
    if (!serializedState) {
      return undefined;
    }

    return JSON.parse(serializedState);
  } catch (error) {
    return undefined;
  }
};

export const persistState: (state: State) => void = state => {
  try {
    if (whiteList.length) {
      const stateToPersist: persistedState = {};

      for (let storeAttr of whiteList) {
        stateToPersist[storeAttr] = JSON.parse(
          JSON.stringify(state[storeAttr as keyof State])
        );
      }

      window.localStorage.setItem(
        PERSISTED_STATE,
        JSON.stringify(stateToPersist)
      );
    }
  } catch (error) {
    console.error("State could not be persisted: ", error);
  }
};
