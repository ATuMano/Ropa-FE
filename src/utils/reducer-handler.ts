import { Action } from "redux";

import { Mapping } from "./types";

export const handle = <S>(
  state: S,
  action: Action,
  map: Mapping<Function>
): S => {
  const handler = map[action.type];
  const hasHandler = !!handler;

  return hasHandler ? handler(state, action) : state;
};
