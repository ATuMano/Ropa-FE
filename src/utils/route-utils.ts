interface Location {
  hostname: string;
  hash?: string;
  origin: string;
  search: string;
  href: string;
  replace(newLocation: string): void;
  assign(newLocation: string): void;
}

const getLocation = (): Location => window.location;

export const getURIOrigin = (): string => getLocation().origin;

export const assignToUrl = (newLocation: string): void => {
  getLocation().assign(newLocation);
};

export const getURLKey = (): string => {
  const { hostname } = getLocation();
  return hostname.split(".")[0];
};

export const useQueryParam = (history: any): URLSearchParams =>
  new URLSearchParams(history.location.search);

export const getQueryParam = (
  query: URLSearchParams,
  paramIdentifier: string
): string | undefined | null => {
  const result = query.get(paramIdentifier);
  switch (result) {
    case "undefined":
      return undefined;
    case "null":
      return null;
    default:
      return result;
  }
};
