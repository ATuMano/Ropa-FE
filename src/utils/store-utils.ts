import { compose } from "redux";

import { isTheSameEnvironmentVariable } from "./variable";

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: Function;
  }
}

export const shouldLogReduxEvents = (): boolean =>
  isTheSameEnvironmentVariable(process.env.REACT_APP_LOGGER, "LOGGER");

export const composeEnhancer: Function =
  (process.env.NODE_ENV === "development" &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;
