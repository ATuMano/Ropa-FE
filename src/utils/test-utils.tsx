import React from "react";
import createSagaMiddleware, { runSaga } from "redux-saga";
import { createStore, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
import {
  render,
  cleanup,
  screen,
  waitFor,
  waitForElementToBeRemoved
} from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { actionWatcherMiddleware, triggeredActions } from "./triggered-actions";
import rootReducer from "../app/store/reducers";
import sagas from "../app/store/root-saga";
import { State } from "app/store/types";
import defaultTheme from "styles/default-theme";
import { ThemeProvider } from "styled-components";

const initialState: Partial<State> = {
  configuration: {
    partnerName: "Test"
  },
  isAppLoaded: false
};

export const cleanUp = () => {
  cleanup();
  triggeredActions.clear();
};

export const getStore = (providedState = initialState) => {
  const sagaMiddleware = createSagaMiddleware();
  const middleware = [actionWatcherMiddleware, sagaMiddleware];

  const createdStore = createStore(
    rootReducer,
    providedState,
    compose(applyMiddleware(...middleware))
  );
  sagaMiddleware.run(sagas);
  return { store: createdStore };
};

export const renderWithProviders = (
  Component: JSX.Element,
  providedState?: Partial<State>
) => {
  const { store } = getStore(providedState);

  const rendered = render(
    <Provider store={store}>
      <ThemeProvider theme={defaultTheme}>
        <Router>{Component}</Router>
      </ThemeProvider>
    </Provider>
  );
  return { ...rendered, store, screen };
};

export const findErrorByName = async (
  inputName: string
): Promise<HTMLElement> => {
  const find = () => screen.queryByTestId(`error-input-${inputName}`);

  await waitFor(() => !!find());

  return find()!;
};

export const waitForErrorRemovedByName = async (
  inputName: string
): Promise<void> => {
  const find = () => screen.queryByTestId(`error-input-${inputName}`);

  await waitForElementToBeRemoved(find);
};

export async function recordSaga(saga: any, initialAction: any) {
  const dispatched: any = [];

  await runSaga(
    {
      dispatch: action => dispatched.push(action)
    },
    saga,
    initialAction
  ).result;

  return dispatched;
}
