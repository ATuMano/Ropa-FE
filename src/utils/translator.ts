import { ConfigModel } from "features/configuration/configuration-types";

export interface TranslatorHook {
  translate: (key: string, allowEmptyTranslation: boolean) => string;
}

class Translator {
  constructor(
    private readonly config: ConfigModel,
    private readonly currentLanguage: string,
    private readonly translator: TranslatorHook
  ) {}

  translate(key: string, allowEmptyTranslation: boolean): string {
    const value = this.translator.translate(key, allowEmptyTranslation);

    if (value === key) {
      return allowEmptyTranslation ? "" : `#${key}#`;
    }

    return value;
  }
}

export default Translator;
