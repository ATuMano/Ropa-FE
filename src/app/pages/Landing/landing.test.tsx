import React from "react";
import { renderWithProviders } from "utils/test-utils";
import Landing from "./landing";

describe("Landing tests", () => {
  const renderLanding = () => renderWithProviders(<Landing />);

  it("should match snapshot", () => {
    const { asFragment } = renderLanding();
    expect(asFragment()).toMatchSnapshot();
  });
});
