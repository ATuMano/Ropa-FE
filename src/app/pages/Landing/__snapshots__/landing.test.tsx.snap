// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-iBkjds hkfcpW"
  >
    <div
      class="sc-iqcoie ivyQmM"
    >
      <label
        class="sc-gKXOVf jMtTcf"
      >
        A donde quieres ir?
      </label>
      <div
        class=" css-1ktriz-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-1c5nzwc-control"
        >
          <div
            class=" css-so14a7-ValueContainer"
          >
            <div
              class=" css-qc6sy-singleValue"
            >
              Barcelona
            </div>
            <div
              class=" css-a2c4u8-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="category"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1vsx5z8-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class="sc-jqUVSM krTeVt"
              data-testid="clear-selection"
            >
              <svg
                fill="none"
                height="8"
                viewBox="0 0 8 8"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.82843 7.77596L4 4.94753L1.17157 7.77596L0.22405 6.82843L3.05248 4.00001L0.22405 1.17158L1.17157 0.224057L4 3.05248L6.82843 0.224057L7.77595 1.17158L4.94752 4.00001L7.77595 6.82843L6.82843 7.77596Z"
                  fill="#0BB3BE"
                />
              </svg>
            </div>
          </div>
        </div>
        <input
          name="category"
          type="hidden"
          value="Barcelona"
        />
      </div>
      <div
        class="rdt sc-kDDrLX ftcuDn"
      >
        <input
          class="form-control"
          placeholder="Fecha inicio viaje"
          type="text"
          value="01/09/2022"
        />
        <div
          class="rdtPicker"
        >
          <div
            class="rdtDays"
          >
            <table>
              <thead>
                <tr>
                  <th
                    class="rdtPrev"
                  >
                    <span>
                      ‹
                    </span>
                  </th>
                  <th
                    class="rdtSwitch"
                    colspan="5"
                    data-value="8"
                  >
                    septiembre 2022
                  </th>
                  <th
                    class="rdtNext"
                  >
                    <span>
                      ›
                    </span>
                  </th>
                </tr>
                <tr>
                  <th
                    class="dow"
                  >
                    lu
                  </th>
                  <th
                    class="dow"
                  >
                    ma
                  </th>
                  <th
                    class="dow"
                  >
                    mi
                  </th>
                  <th
                    class="dow"
                  >
                    ju
                  </th>
                  <th
                    class="dow"
                  >
                    vi
                  </th>
                  <th
                    class="dow"
                  >
                    sá
                  </th>
                  <th
                    class="dow"
                  >
                    do
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    class="rdtDay rdtOld"
                    data-month="7"
                    data-value="29"
                    data-year="2022"
                  >
                    29
                  </td>
                  <td
                    class="rdtDay rdtOld"
                    data-month="7"
                    data-value="30"
                    data-year="2022"
                  >
                    30
                  </td>
                  <td
                    class="rdtDay rdtOld"
                    data-month="7"
                    data-value="31"
                    data-year="2022"
                  >
                    31
                  </td>
                  <td
                    class="rdtDay rdtActive"
                    data-month="8"
                    data-value="1"
                    data-year="2022"
                  >
                    1
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="2"
                    data-year="2022"
                  >
                    2
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="3"
                    data-year="2022"
                  >
                    3
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="4"
                    data-year="2022"
                  >
                    4
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="5"
                    data-year="2022"
                  >
                    5
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="6"
                    data-year="2022"
                  >
                    6
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="7"
                    data-year="2022"
                  >
                    7
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="8"
                    data-year="2022"
                  >
                    8
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="9"
                    data-year="2022"
                  >
                    9
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="10"
                    data-year="2022"
                  >
                    10
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="11"
                    data-year="2022"
                  >
                    11
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="12"
                    data-year="2022"
                  >
                    12
                  </td>
                  <td
                    class="rdtDay rdtToday"
                    data-month="8"
                    data-value="13"
                    data-year="2022"
                  >
                    13
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="14"
                    data-year="2022"
                  >
                    14
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="15"
                    data-year="2022"
                  >
                    15
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="16"
                    data-year="2022"
                  >
                    16
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="17"
                    data-year="2022"
                  >
                    17
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="18"
                    data-year="2022"
                  >
                    18
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="19"
                    data-year="2022"
                  >
                    19
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="20"
                    data-year="2022"
                  >
                    20
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="21"
                    data-year="2022"
                  >
                    21
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="22"
                    data-year="2022"
                  >
                    22
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="23"
                    data-year="2022"
                  >
                    23
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="24"
                    data-year="2022"
                  >
                    24
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="25"
                    data-year="2022"
                  >
                    25
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="26"
                    data-year="2022"
                  >
                    26
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="27"
                    data-year="2022"
                  >
                    27
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="28"
                    data-year="2022"
                  >
                    28
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="29"
                    data-year="2022"
                  >
                    29
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="30"
                    data-year="2022"
                  >
                    30
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="1"
                    data-year="2022"
                  >
                    1
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="2"
                    data-year="2022"
                  >
                    2
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="3"
                    data-year="2022"
                  >
                    3
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="4"
                    data-year="2022"
                  >
                    4
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="5"
                    data-year="2022"
                  >
                    5
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="6"
                    data-year="2022"
                  >
                    6
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="7"
                    data-year="2022"
                  >
                    7
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="8"
                    data-year="2022"
                  >
                    8
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="9"
                    data-year="2022"
                  >
                    9
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        class="rdt sc-kDDrLX ftcuDn"
      >
        <input
          class="form-control"
          placeholder="Fecha fin viaje"
          type="text"
          value="11/09/2022"
        />
        <div
          class="rdtPicker"
        >
          <div
            class="rdtDays"
          >
            <table>
              <thead>
                <tr>
                  <th
                    class="rdtPrev"
                  >
                    <span>
                      ‹
                    </span>
                  </th>
                  <th
                    class="rdtSwitch"
                    colspan="5"
                    data-value="8"
                  >
                    septiembre 2022
                  </th>
                  <th
                    class="rdtNext"
                  >
                    <span>
                      ›
                    </span>
                  </th>
                </tr>
                <tr>
                  <th
                    class="dow"
                  >
                    lu
                  </th>
                  <th
                    class="dow"
                  >
                    ma
                  </th>
                  <th
                    class="dow"
                  >
                    mi
                  </th>
                  <th
                    class="dow"
                  >
                    ju
                  </th>
                  <th
                    class="dow"
                  >
                    vi
                  </th>
                  <th
                    class="dow"
                  >
                    sá
                  </th>
                  <th
                    class="dow"
                  >
                    do
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    class="rdtDay rdtOld"
                    data-month="7"
                    data-value="29"
                    data-year="2022"
                  >
                    29
                  </td>
                  <td
                    class="rdtDay rdtOld"
                    data-month="7"
                    data-value="30"
                    data-year="2022"
                  >
                    30
                  </td>
                  <td
                    class="rdtDay rdtOld"
                    data-month="7"
                    data-value="31"
                    data-year="2022"
                  >
                    31
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="1"
                    data-year="2022"
                  >
                    1
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="2"
                    data-year="2022"
                  >
                    2
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="3"
                    data-year="2022"
                  >
                    3
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="4"
                    data-year="2022"
                  >
                    4
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="5"
                    data-year="2022"
                  >
                    5
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="6"
                    data-year="2022"
                  >
                    6
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="7"
                    data-year="2022"
                  >
                    7
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="8"
                    data-year="2022"
                  >
                    8
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="9"
                    data-year="2022"
                  >
                    9
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="10"
                    data-year="2022"
                  >
                    10
                  </td>
                  <td
                    class="rdtDay rdtActive"
                    data-month="8"
                    data-value="11"
                    data-year="2022"
                  >
                    11
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="12"
                    data-year="2022"
                  >
                    12
                  </td>
                  <td
                    class="rdtDay rdtToday"
                    data-month="8"
                    data-value="13"
                    data-year="2022"
                  >
                    13
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="14"
                    data-year="2022"
                  >
                    14
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="15"
                    data-year="2022"
                  >
                    15
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="16"
                    data-year="2022"
                  >
                    16
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="17"
                    data-year="2022"
                  >
                    17
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="18"
                    data-year="2022"
                  >
                    18
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="19"
                    data-year="2022"
                  >
                    19
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="20"
                    data-year="2022"
                  >
                    20
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="21"
                    data-year="2022"
                  >
                    21
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="22"
                    data-year="2022"
                  >
                    22
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="23"
                    data-year="2022"
                  >
                    23
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="24"
                    data-year="2022"
                  >
                    24
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="25"
                    data-year="2022"
                  >
                    25
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="26"
                    data-year="2022"
                  >
                    26
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="27"
                    data-year="2022"
                  >
                    27
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="28"
                    data-year="2022"
                  >
                    28
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="29"
                    data-year="2022"
                  >
                    29
                  </td>
                  <td
                    class="rdtDay"
                    data-month="8"
                    data-value="30"
                    data-year="2022"
                  >
                    30
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="1"
                    data-year="2022"
                  >
                    1
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="2"
                    data-year="2022"
                  >
                    2
                  </td>
                </tr>
                <tr>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="3"
                    data-year="2022"
                  >
                    3
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="4"
                    data-year="2022"
                  >
                    4
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="5"
                    data-year="2022"
                  >
                    5
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="6"
                    data-year="2022"
                  >
                    6
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="7"
                    data-year="2022"
                  >
                    7
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="8"
                    data-year="2022"
                  >
                    8
                  </td>
                  <td
                    class="rdtDay rdtNew"
                    data-month="9"
                    data-value="9"
                    data-year="2022"
                  >
                    9
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <button
        class="sc-gsnTZi fssMSn"
      >
        BUSCAR
      </button>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-1 sc-evZas bxNdLL"
    data-testid="landing-imagen-container"
  >
    <div
      class="MuiBox-root MuiBox-root-2 sc-breuTD eWFjNK"
      data-testid="landing-text-container"
    >
      <h4
        class="MuiTypography-root sc-ksZaOG JypUJ MuiTypography-h4"
        data-testid="landing-title"
      >
        Disfruta tus vacas
      </h4>
      <p
        class="MuiTypography-root sc-ksZaOG JypUJ MuiTypography-body2 MuiTypography-paragraph"
        data-testid="landing-paragraph-1"
      >
        Sabemos el estres de armar la valija. Es por eso que te solucionamos para que puedas enfocarte en disfrutar!
      </p>
      <p
        class="MuiTypography-root sc-ksZaOG JypUJ MuiTypography-body2 MuiTypography-paragraph"
        data-testid="landing-paragraph-2"
      >
        Preparate el pasaporte y el itinerario del viaje. Nosotros nos ocupamos de la ropa!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
